SU2_CFD/include/numerics/turbulent/transition/trans_sources.hpp

```diff
@@ -0,0 +1,290 @@
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "../../../../Common/include/toolboxes/geometry_toolbox.hpp"
+#include "../../scalar/scalar_sources.hpp"
+
+
+/*!
+ * \class CSourcePieceWise_TranLM
+ * \brief Class for integrating the source terms of the Menter SST turbulence model equations.
+ * \ingroup SourceDiscr
+ * \author S. Kang.
+ */
+template <class FlowIndices>
+class CSourcePieceWise_TransLM final : public CNumerics {
+ private:
+  const FlowIndices idx; /*!< \brief Object to manage the access to the flow primitives. */
+
+  su2double g_eff_i,
+  g_eff_j,
+  g_sep_i,
+  g_sep_j;
+  /*--- LM Closure constants ---*/
+  const su2double c_e1 = 1.0;
+  const su2double c_a1 = 2.0;
+  const su2double c_e2 = 50.0;
+  const su2double c_a2 = 0.06;
+  const su2double sigmaf = 1.0;
+  const su2double s1 = 2.0;
+  const su2double c_theta = 0.03;
+  const su2double sigmat = 2.0;
+
+  su2double Vorticity;
+  su2double Residual[2];
+  su2double* Jacobian_i[2];
+  su2double Jacobian_Buffer[4]; /// Static storage for the Jacobian (which needs to be pointer for return type).
+
+ public:
+  /*!
+   * \brief Constructor of the class.
+   * \param[in] val_nDim - Number of dimensions of the problem.
+   * \param[in] constants - SST model constants.
+   * \param[in] val_kine_Inf - Freestream k, for SST with sustaining terms.
+   * \param[in] val_omega_Inf - Freestream w, for SST with sustaining terms.
+   * \param[in] config - Definition of the particular problem.
+   */
+  CSourcePieceWise_TransLM(unsigned short val_nDim, unsigned short val_nVar, const CConfig* config) 
+      : CNumerics(val_nDim, 2, config),
+        idx(val_nDim, config->GetnSpecies()) {
+
+    
+    /*--- "Allocate" the Jacobian using the static buffer. ---*/
+    Jacobian_i[0] = Jacobian_Buffer;
+    Jacobian_i[1] = Jacobian_Buffer + 2;
+  }
+
+  /*!
+   * \brief Residual for source term integration.
+   * \param[in] config - Definition of the particular problem.
+   * \return A lightweight const-view (read-only) of the residual/flux and Jacobians.
+   */
+  ResidualType<> ComputeResidual(const CConfig* config) override {
+  /*--- ScalarVar[0] = k, ScalarVar[0] = w, TransVar[0] = gamma, and TransVar[0] = ReThetaT ---*/
+  /*--- dU/dx = PrimVar_Grad[1][0] ---*/
+  AD::StartPreacc();
+  AD::SetPreaccIn(StrainMag_i);
+  AD::SetPreaccIn(ScalarVar_i, nVar);
+  AD::SetPreaccIn(ScalarVar_Grad_i, nVar, nDim);
+  AD::SetPreaccIn(TransVar_i, nVar);
+  AD::SetPreaccIn(TransVar_Grad_i, nVar, nDim);
+  AD::SetPreaccIn(Volume); AD::SetPreaccIn(dist_i);
+  AD::SetPreaccIn(&V_i[idx.Velocity()], nDim);
+  AD::SetPreaccIn(PrimVar_Grad_i, nDim+idx.Velocity(), nDim);
+  AD::SetPreaccIn(Vorticity_i, 3);
+  
+
+  unsigned short iDim;
+  
+  su2double Corr_Rec, Corr_F_length, Corr_Ret, F_length;
+  su2double F_onset1, F_onset2, F_onset3, F_onset;
+  su2double f_turb, Re_v, f_sub, r_omega;
+  su2double Pg, Dg, Pthetat;
+  su2double diverg;
```

Comment:
Introduce these where they are used please.

Reply:
This is NASA turbulence modeling resources of LM model (TMR) [SST-2003-LM2009](https://turbmodels.larc.nasa.gov/langtrymenter_4eqn.html) 

All variables are almost the same in Langtry's [paper](https://doi.org/10.2514/1.42362) and [dissertation.](https://elib.uni-stuttgart.de/handle/11682/1722)


F_onset1, F_onset2, Fonset3, and F_onset of line 105 is the transition onset function that triggers the transition.


f_turb is used to disable the destruction/relaminarization source outside of a laminar boundary layer or in the viscous sublayer and is defined as follows :  
![image](https://user-images.githubusercontent.com/50565314/185124203-b9be3e98-0e98-49d4-bae0-b776a6a8d3d0.png)

Re_v  is vorticity Reynolds number : 
![image](https://user-images.githubusercontent.com/50565314/185129854-92a2e70a-3ac1-4b75-8bd2-505848e5ff20.png)

f_sub is forcing F_length to always be equal to its maximum value (in this case 40.0) in the viscous sublayer : 
![image](https://user-images.githubusercontent.com/50565314/185140667-47eafe65-94b0-4010-9ae7-394dfd969834.png)

r_omega is Reynolds number base on omega : 
![image](https://user-images.githubusercontent.com/50565314/185144767-516d6a5b-7473-4a52-8a5c-6256a40cf07c.png)


 Pg, Dg, and Pthetat are production and destruction terms of each transport equation.

diverg is divergence of velocity.

Reply:
Sorry, I mean instead of mass declaring the variables at the top of a function, declare the variables when they are used ->

i.e. "su2double F_onset = ......"